rust/core-lib/src/editor.rs

```diff
@@ -941,6 +942,17 @@ impl Editor {
 
         Some(GetDataResponse { chunk, offset, first_line, first_line_offset })
     }
+
+    pub(crate) fn pre_save(&mut self, view: &mut View, config: &BufferItems,) {
+        if let Some(last_line) = self.text.lines_raw(Interval::new(0,2)).last() {
+            eprintln!("Last line: {}", last_line);
+            if let Ok(line_ending) = LineEnding::parse_chunk(&last_line) {
+                if line_ending.is_none() {
+                    self.insert_newline(view, config)
```

Comment:
For this to work I need to get view&config passed into pre_save, which isn't very pretty. Also, I guess this inserts the newline at the current cursor position?